pkg/client/optimizely_user_context_test.go

```diff
@@ -1221,6 +1221,26 @@ func (s *OptimizelyUserContextTestSuite) TestForcedDecision() {
 	s.Error(err)
 }
 
+func (s *OptimizelyUserContextTestSuite) TestDecideAllFlagsWithBatchUPS() {
```

Comment:
Would it be good to add coverage for `DecideForKeysWithBatchUPS` and `DecideWithBatchUPS`?  

It may also be good to cover the expected contents of the args going into `Lookup` & `Save`.

Reply:
added

Reply:
Updated unit tests